jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -5958,20 +5958,10 @@ public int indexOf(int c, int start) {
 				// Check if the String is compressed
 				if (enableCompression && (null == compressionFlag || count >= 0)) {
 					if (c <= 255) {
-						byte b = (byte) c;
-
-						for (int i = start; i < len; ++i) {
-							if (helpers.getByteFromArrayByIndex(array, i) == b) {
-								return i;
-							}
-						}
+						return helpers.findElementFromArray(array, (byte)c, start, len);
```

Comment:
There are two implementations of `public int indexOf(int c, int start)` in this file. One for Java 8 and one for Java 9+. Could we make the same change for the Java 9+ version?

Reply:
Sorry I didn't realized that the two functions now are separated. I've updated both of them.